ToolBox/Benzinga/BenzingaNewsDataConverter.cs

```diff
@@ -0,0 +1,451 @@
+                    {
+                        symbolsToRemove.Add(symbol);
+                        Log.Error($"BenzingaDataConverter.SerializeArticlesAndIndexes(): Ticker {symbol.Value} contains invalid character ':'. Skipping");
+                        continue;
+                    }
+
+                    // We can't write these tickers in Windows, so we'll skip them
+                    if (_isWindows && _forbiddenTickers.Contains(symbol.Value))
+                    {
+                        symbolsToRemove.Add(symbol);
+                        Log.Error($"BenzingaDataConverter.SerializeArticlesAndIndexes(): Ticker {symbol.Value} is invalid in Windows. Skipping");
+                        continue;
+                    }
+
+                    BenzingaIndexCollection indexCollection;
+                    if (!filtered.SymbolIndexes.TryGetValue(symbol, out indexCollection))
+                    {
+                        indexCollection = new BenzingaIndexCollection(article.UpdatedAt.Date);
+                        filtered.SymbolIndexes[symbol] = indexCollection;
+                        // The reference file path is where the indexes pointing to an article live
+                        var referenceFile = new FileInfo(Path.Combine(_processedFilesDirectory.FullName, symbol.Value.ToLowerInvariant(), $"{article.UpdatedAt.Date:yyyyMMdd}.csv"));
+
+                        if (referenceFile.Exists)
+                        {
+                            foreach (var reference in File.ReadLines(referenceFile.FullName))
+                            {
+                                // Because `indexCollection.Indexes` is a HashSet, we're guaranteed
+                                // to never have duplicate values after running the converter, even
+                                // if we somehow previously had duplicate values in the index file.
+                                indexCollection.Indexes.Add(reference);
+                            }
+                        }
+                    }
+
+                    // Add the article ID to the index collection so that we can batch
+                    // write all of the indexes in one go per day
+                    indexCollection.Indexes.Add($"{article.Id}.json");
+
+                    createdReference = true;
+                }
+
+                // Skip if we didn't create any references for an article
+                if (!createdReference)
+                {
+                    Log.Error($"BenzingaDataConverter.SerializeArticlesAndIndexes(): Skipping news article {article.Id} because no tickers are associated with it");
+                    continue;
+                }
+
+                // Remove unwanted symbols from the article before we write to disk
+                foreach (var symbolToRemove in symbolsToRemove)
+                {
+                    article.Symbols = article.Symbols.Where(s => s != symbolToRemove).ToList();
```

Comment:
nit - I think it might be more efficient to directly call `.Symbols.Remove(symbolToRemove)`